### PR TITLE
Periodic shard sync metric

### PR DESCRIPTION
**Pull request by ychunxue**
*Issue #, if available:*

*Description of changes:*

Adding metrics and configs for Periodic shard sync

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java [hunk 1]
```diff
@@ -175,6 +175,16 @@
      */
     public static final ShardSyncStrategyType DEFAULT_SHARD_SYNC_STRATEGY_TYPE = ShardSyncStrategyType.SHARD_END;
 
+    /**
+     * Default Lease Recovery Auditor execution period for SHARD_END ShardSyncStrategyType.
+     */
+    public static final long LEASES_RECOVERY_AUDITOR_EXECUTION_FREQUENCY_MILLIS = 2 * 60 * 1000L;
+
+    /**
+     * Default Lease Recovery Auditor inconsistency confidence threshold for running full shard sync for HARD_END ShardSyncStrategyType.
```
Comment (joshua-kim) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java:184]: SHARD_END

Comment (joshua-kim) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java]: SHARD_END

**Review by joshua-kim — APPROVED** [review]
LGTM

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java (hunk 1 above)
Comment (ychunxue) [reply src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java]: Addressed

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java [hunk 2]
```diff
@@ -166,17 +175,27 @@ private void runShardSync() {
         if (leaderDecider.isLeader(workerId)) {
             LOG.debug("WorkerId " + workerId + " is a leader, running the shard sync task");
 
+            MetricsHelper.startScope(metricsFactory, PERIODIC_SHARD_SYNC_MANAGER);
+            boolean isRunSuccess = false;
+            final long runStartMillis = System.currentTimeMillis();
+
             try {
                 final ShardSyncResponse shardSyncResponse = checkForShardSync();
+                MetricsHelper.getMetricsScope().addData("ShouldDoShardSync", shardSyncResponse.shouldDoShardSync() ? 1 : 0, StandardUnit.Count, MetricsLevel.SUMMARY);
```
Comment (ashwing) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java]: InitiatingShardSync

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java [hunk 3]
```diff
@@ -166,17 +175,27 @@ private void runShardSync() {
...
+                MetricsHelper.getMetricsScope().addData("HashRangeHoleDetected", shardSyncResponse.isHoleDetected() ? 1 : 0, StandardUnit.Count, MetricsLevel.SUMMARY);
```
Comment (ashwing) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java]: DetectedIncompleteLease

**Review by ashwing — COMMENTED** [review]
left comments on metric names. Looks good otherwise.

**Review by ashwing — APPROVED** [review]
lgtm

**Pull request merged**